extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -262,6 +260,24 @@ export class Bind {
   }
 
   /**
+   * Returns all descendant elements in a given node.
+   * @param {!Node} node
+   * @return {!Array<Element>}
+   * @private
+   */
+  elementsInNode_(node) {
+    const doc = dev().assert(node.ownerDocument, 'ownerDocument is null.');
+    const treeWalker = doc.createTreeWalker(node, NodeFilter.SHOW_ELEMENT);
+
+    const elements = [];
+    let currentElement;
+    while (currentElement = treeWalker.nextNode()) {
+      elements.push(currentElement);
+    }
```

Comment:
So the great part about `TreeWalker` is you don't need to greedily grab every element into an array. Let's pass around the `walker` instance, and grab `#nextNode` in the [`chunktion ` loops](https://github.com/ampproject/amphtml/pull/7288/files#diff-6a804af8fcfeafe72321b2a9c0f8f050R194).